qiskit/providers/ibmq/api/session.py

```diff
@@ -259,3 +268,50 @@ def _modify_chained_exception_messages(self, exc: BaseException) -> None:
                 exc_message = exc_message.replace(self.access_token, '...')
             modified_args.append(exc_message)
         exc.args = tuple(modified_args)
+
+    def _log_request_info(
+            self,
+            url: str,
+            bare: bool,
+            method: str,
+            request_data: Dict[str, Any]
+    ) -> None:
+        """Log the request data, filtering out specific information.
+
+        Note:
+            The string ``...`` is used to denote information that has been filtered out
+            from the request, within the url and request data.
+
+            The following endpoint URLs are not logged in order to reduce noise: ``/users``
+            and ``/version``. Likewise, the request data is only logged for the following
+            URLs, since they contain useful information: ``/Jobs`` (POST),
+            ``/Jobs/status`` (GET), and ``/devices/<device_name>/properties`` (GET).
+
+            Currently, the backend name is filtered out from endpoint URLs, using a regex
+            to capture the name, and from the data sent to the server when submitting a job.
+        """
+        # Replace the device name in the URL with `...` if it matches, otherwise leave it as is.
+        filtered_url = re.sub(RE_DEVICES_ENDPOINT, '\\1...\\3', url)
+
+        if (not filtered_url.startswith(('/users', '/version'))
+                and filtered_url not in ('/devices/.../queue/status', '/devices/v/1')):
+            try:
+                method = method.upper()
+
+                if filtered_url in ('/Jobs/status', '/devices/.../properties', '/Jobs'):
+                    if filtered_url == '/Jobs' and 'json' in request_data:
+                        # Replace backend name in job submission request.
+                        request_data['json']['backend']['name'] = '...'
+                    logger.debug('Endpoint: %s. Method: %s. Request Data: %s.',
+                                 filtered_url, method, request_data)
+                else:
+                    if bare:
+                        if method == 'PUT':
+                            logger.debug('Uploading Qobj to object storage.')
+                        elif method == 'GET':
+                            logger.debug('Downloading Qobj from object storage.')
```

Comment:
Can you move these loggings to the actual job/result upload/download methods? I think it's better not to rely on PUT/GET with bare url to be object storage upload/download.